### PR TITLE
Some cleanup of AnkiDroidApp.java

### PR DESCRIPTION
**Pull request by hssm**
Here is some work to slim down this class and add some comments to it.

I've deleted dead code, moved out parts that didn't belong in it, fixed some warnings, and factored out the AnkiDroid directory creation routines. The idea was to ensure there is exactly one place in the application where we create the AnkiDroid directory and/or check it for write permission to make it easeir later to handle all those crashes we're getting from pointing to inaccessible directories.

This also fixes crashes where the user sets the AnkiDroid directory to a blank string, but not any other kind of inaccessible path.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidApp.java [hunk 1]
```diff
@@ -224,55 +217,44 @@ public void onCreate() {
...
             DEFAULT_SWIPE_MIN_DISTANCE = vc.getScaledPagingTouchSlop();
         } else {
             DEFAULT_SWIPE_MIN_DISTANCE = vc.getScaledTouchSlop()*2;
         }
         DEFAULT_SWIPE_THRESHOLD_VELOCITY = vc.getScaledMinimumFlingVelocity();
+
+        // Create the AnkiDroid directory if missing
+        initializeAnkiDroidDirectory();
+        // TODO: Check for write access before proceeding
+        // We should not allow the application to proceed if this directory is inaccessible.
```
Comment (hssm) [inline AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidApp.java]: So somewhere here we can show an error dialog and let the user pick another directory to use if the currently specified one isn't accessible.

File: AnkiDroid/src/main/java/com/ichi2/anki/NavigationDrawerActivity.java [hunk 2]
```diff
@@ -353,6 +353,8 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                 // collection path has changed so kick the user back to the DeckPicker
                 AnkiDroidApp.closeCollection(true);
                 finishWithoutAnimation();
+                // Ensure the new collection directory exists before reloading
+                AnkiDroidApp.initializeAnkiDroidDirectory();
```
Comment (hssm) [inline AnkiDroid/src/main/java/com/ichi2/anki/NavigationDrawerActivity.java]: The ability to change the AnkiDroid directory at runtime means there are actually two places where we have to worry about the accessibility of the AnkiDroid directory.

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidApp.java (hunk 1 above)
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/AnkiDroidApp.java]: Probably easier to just send an intent to open settings

**Comment by timrae** [comment]
Thanks for this. I think we should probably include the changes in a point release as well, since they will help with debugging

**Comment by hssm** [comment]
That's probably a good idea. I'll do a bit more work on this permission error thing. There have been a lot of them lately.

You've probably noticed the abundance of `Could not open database` and such crashes on the crash reporter but these are all permission problems. The logcat snippet (which is *so* useful) has lines like these:

```
02-01 15:23:08.866 E/AnkiDroid( 5011): Storage/ Nomedia file could not be created
02-01 15:23:08.866 E/AnkiDroid( 5011): AnkiDroidApp/ Nomedia file could not be created in path /storage/emulated/0/AnkiDroid```

Which basically tells us there is no write access (even though the directory itself might exist).

**Pull request merged**